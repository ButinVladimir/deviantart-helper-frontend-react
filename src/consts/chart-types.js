export const CHART_VIEWS = 'views';
export const CHART_FAVOURITES = 'favourites';
export const CHART_COMMENTS = 'comments';
export const CHART_DOWNLOADS = 'downloads';
