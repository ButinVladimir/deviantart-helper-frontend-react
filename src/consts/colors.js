const LIGHT_BLUE = '#88f';
const RED = '#b33';
const GREEN = '#393';
const BLUE = '#11c';
const ORANGE = '#f94';
const PURPLE = '#94a';
const CYAN = '#2b9';
const BLACK = '#111';
const GREY = '#999';
const PINK = '#d5b';

export default [LIGHT_BLUE, RED, GREEN, BLUE, ORANGE, PURPLE, CYAN, BLACK, GREY, PINK];
