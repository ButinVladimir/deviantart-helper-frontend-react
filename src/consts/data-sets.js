export const DATA_SET_TIMESTAMP = 0;
export const DATA_SET_VIEWS = 1;
export const DATA_SET_FAVOURITES = 2;
export const DATA_SET_COMMENTS = 3;
export const DATA_SET_DOWNLOADS = 4;

export const dataSetOptions = [
  [DATA_SET_VIEWS, 'Views'],
  [DATA_SET_FAVOURITES, 'Favourites'],
  [DATA_SET_COMMENTS, 'Comments'],
  [DATA_SET_DOWNLOADS, 'Downloads'],
];
