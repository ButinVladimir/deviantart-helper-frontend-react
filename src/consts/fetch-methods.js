export const GET = 'GET';
export const POST = 'POST';
