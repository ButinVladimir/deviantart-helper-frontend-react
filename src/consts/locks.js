export const LOCK_LOAD_USER_INFO = 'LOCK_LOAD_USER_INFO';
export const LOCK_START_LOADING_DATA = 'LOCK_START_LOADING_DATA';
export const LOCK_REVOKE = 'LOCK_REVOKE';

export const LOCK_BROWSE_DEVIATIONS = 'LOCK_BROWSE_DEVIATIONS';
export const LOCK_DEVIATION_DETAILS = 'LOCK_DEVIATION_DETAILS';
export const LOCK_DEVIATION_DETAILS_METADATA = 'LOCK_DEVIATION_DETAILS_METADATA';
export const LOCK_DEVIATIONS_STATISTICS = 'LOCK_DEVIATIONS_STATISTICS';
export const LOCK_DEVIATIONS_TOTAL = 'LOCK_DEVIATIONS_TOTAL';
