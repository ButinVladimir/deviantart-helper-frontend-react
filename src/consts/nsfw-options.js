export const SHOW_ALL = 'SHOW_ALL';
export const SHOW_NON_NSFW = 'SHOW_NON_NSFW';
export const SHOW_NSFW = 'SHOW_NSFW';

export const nsfwOptions = [
  [SHOW_ALL, 'Show everything'],
  [SHOW_NON_NSFW, 'Show non-NSFW'],
  [SHOW_NSFW, 'Show NSFW'],
];
