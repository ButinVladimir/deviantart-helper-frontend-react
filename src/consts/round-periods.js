export const ROUND_PERIOD_NO_ROUND = 1;
export const ROUND_PERIOD_1_HOUR = 60 * 60 * 1000;
export const ROUND_PERIOD_4_HOURS = 4 * ROUND_PERIOD_1_HOUR;
export const ROUND_PERIOD_12_HOURS = 12 * ROUND_PERIOD_1_HOUR;
export const ROUND_PERIOD_1_DAY = 24 * ROUND_PERIOD_1_HOUR;

export const roundPeriodOptions = [
  [ROUND_PERIOD_NO_ROUND, 'No rounding'],
  [ROUND_PERIOD_1_HOUR, '1 hour'],
  [ROUND_PERIOD_4_HOURS, '4 hours'],
  [ROUND_PERIOD_12_HOURS, '12 hours'],
  [ROUND_PERIOD_1_DAY, '1 day'],
];
