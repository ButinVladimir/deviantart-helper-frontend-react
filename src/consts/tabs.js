export const DESCRIPTION = 'description';
export const PREVIEW = 'preview';
export const CHART = 'chart';
export const LIST = 'list';

export const deviationDetailsTabs = [
  [DESCRIPTION, 'Description'],
  [PREVIEW, 'Preview'],
  [CHART, 'Chart'],
];

export const deviationStatisticsTabs = [
  [LIST, 'List'],
  [CHART, 'Chart'],
];
