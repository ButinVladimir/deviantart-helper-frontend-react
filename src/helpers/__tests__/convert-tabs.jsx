import React from 'react';
import Tabs from 'react-bulma-components/lib/components/tabs';
import { shallow } from 'enzyme';
import convertTabs from '../convert-tabs';
import { deviationDetailsTabs, PREVIEW, CHART } from '../../consts/tabs';

describe('Convert tabs helper', () => {
  it('can render correctly', () => {
    const wrapper = shallow(
      <div>
        {convertTabs(deviationDetailsTabs, PREVIEW, () => {})}
      </div>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('can handle tab select', () => {
    const handler = jest.fn();
    const wrapper = shallow(
      <div>
        {convertTabs(deviationDetailsTabs, PREVIEW, tab => () => handler(tab))}
      </div>,
    );

    wrapper.find(Tabs.Tab).at(2).simulate('click');

    expect(handler).toHaveBeenCalledTimes(1);
    expect(handler).toHaveBeenCalledWith(CHART);
  });
});
