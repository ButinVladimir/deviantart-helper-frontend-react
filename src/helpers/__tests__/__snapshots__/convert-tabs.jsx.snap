// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Convert tabs helper can render correctly 1`] = `
<div>
  <Tab
    active={false}
    className=""
    key="description"
    renderAs="a"
    style={Object {}}
  >
    Description
  </Tab>
  <Tab
    active={true}
    className=""
    key="preview"
    renderAs="a"
    style={Object {}}
  >
    Preview
  </Tab>
  <Tab
    active={false}
    className=""
    key="chart"
    renderAs="a"
    style={Object {}}
  >
    Chart
  </Tab>
</div>
`;
